### PR TITLE
adding notification facet

### PR DESCRIPTION
**Pull request by truhacevkir**
Aggregation return following notifications:
```

{
  "notifications": [
    {
      "id": "https://api.sandbox.nva.aws.unit.no/search/customer/tickets?aggregation=all&size=0&from=0&notifications=DoiRequestNotification",
      "key": "DoiRequestNotification",
      "count": 0
    },
    {
      "id": "https://api.sandbox.nva.aws.unit.no/search/customer/tickets?aggregation=all&size=0&from=0&notifications=UnassignedNotification",
      "key": "UnassignedNotification",
      "count": 8
    },
    {
      "id": "https://api.sandbox.nva.aws.unit.no/search/customer/tickets?aggregation=all&size=0&from=0&notifications=GeneralSupportNotification",
      "key": "GeneralSupportNotification",
      "count": 2
    },
    {
      "id": "https://api.sandbox.nva.aws.unit.no/search/customer/tickets?aggregation=all&size=0&from=0&notifications=UserNotification",
      "key": "UserNotification",
      "count": 3
    },
    {
      "id": "https://api.sandbox.nva.aws.unit.no/search/customer/tickets?aggregation=all&size=0&from=0&notifications=PublishingRequestNotification",
      "key": "PublishingRequestNotification",
      "count": 1
    }
  ]
}
        
```
        
Searching these notifications does not work.

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java [hunk 1]
```diff
@@ -34,15 +36,37 @@ public static JsonNode apply(JsonNode aggregations, Map<String, String> definiti
...
+                keysToRemove.add(fieldName);
+            }
+        }
+        keysToRemove.forEach(jsonNode::remove);
+        jsonNode.put("notifications", notifications);
+    }
+
+    private static boolean isNotificationAggregation(Entry<String, JsonNode> field) {
+        return field.getKey().toLowerCase(Locale.getDefault()).contains("notifications") && field.getValue().isObject();
+    }
```
Comment (truhacevkir) [inline search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java:64]: This is ugly, it should be possible to apply to Ticket aggregations only, but i feel this is the right place to do it. We can consider to create class TicketAggregation which would implement Aggregation and "apply" own rules to it.

File: search-commons/src/main/java/no/unit/nva/search2/ticket/Constants.java [hunk 2]
```diff
@@ -37,51 +39,192 @@ public final class Constants {
...
+                                                                                       + PUBLICATION
+                                                                                       + DOT
+                                                                                       + IDENTIFIER
+                                                                                       + DOT
+                                                                                       + KEYWORD;
     public static final String PUBLICATION_MAIN_TITLE_KEYWORD = PUBLICATION + DOT + MAIN_TITLE + DOT + KEYWORD;
     public static final String PUBLICATION_MODIFIED_DATE = PUBLICATION + DOT + MODIFIED_DATE;
     public static final String PUBLICATION_OWNER_KEYWORD = PUBLICATION + DOT + OWNER_KEYWORD;
     public static final String PUBLICATION_STATUS_KEYWORD = PUBLICATION + DOT + STATUS_KEYWORD;
+    public static final String MESSAGE_FIELDS = MESSAGES
```
Comment (truhacevkir) [inline search-commons/src/main/java/no/unit/nva/search2/ticket/Constants.java:57]: Code style does this (NVA)

File: search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java (hunk 1 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java]: I tend to agree with  you, would you like to refactoring it?

Comment (truhacevkir) [reply search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java]: Yes, but it another pr.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**